packages/core/test/session.test.ts

```diff
@@ -310,6 +314,203 @@ describe('session', () => {
+        ])
+      })
+      const connection = mockBeginWithSuccess(newFakeConnection())
+
+      const { session, connectionProvider } = setupSession({
+        connection,
+        bookmarkManager: manager,
+        beginTx: false,
+        lastBookmarks: new bookmarks.Bookmarks(customBookmarks),
+        database: 'neo4j'
+      })
+
+      await session.beginTransaction()
+
+      expect(connectionProvider.acquireConnection).toBeCalledWith(
+        expect.objectContaining({ bookmarks: new bookmarks.Bookmarks([...customBookmarks, ...systemBookmarks]) })
+      )
+    })
+
+    it.each([
+      [[]],
+      [customBookmarks]
+    ])('should call getAllBookmarks for the relevant database', async (lastBookmarks) => {
+      const manager = bookmarkManager({
+        initialBookmarks: new Map([
+          ['neo4j', neo4jBookmarks],
+          ['system', systemBookmarks]
+        ])
+      })
+      const getAllBookmarksSpy = jest.spyOn(manager, 'getAllBookmarks')
+
+      const connection = mockBeginWithSuccess(newFakeConnection())
+
+      const { session } = setupSession({
+        connection,
+        bookmarkManager: manager,
+        beginTx: false,
+        database: 'neo4j',
+        lastBookmarks: new bookmarks.Bookmarks(lastBookmarks)
+      })
+
+      await session.beginTransaction()
+
+      expect(getAllBookmarksSpy).toBeCalledWith()
+    })
+
+    it.each([
+      [[]],
+      [customBookmarks]
+    ])('should call begin query with getAllBookmarks + lastBookmarks', async (lastBookmarks) => {
+      const manager = bookmarkManager({
+        initialBookmarks: new Map([
+          ['neo4j', neo4jBookmarks],
+          ['system', systemBookmarks]
+        ])
+      })
+
+      const connection = mockBeginWithSuccess(newFakeConnection())
+
+      const { session } = setupSession({
+        connection,
+        bookmarkManager: manager,
+        beginTx: false,
+        database: 'neo4j',
+        lastBookmarks: new bookmarks.Bookmarks(lastBookmarks)
+      })
+
+      await session.beginTransaction()
+
+      expect(connection.seenBeginTransaction[0]).toEqual([
+        expect.objectContaining({
+          bookmarks: new bookmarks.Bookmarks([...neo4jBookmarks, ...systemBookmarks, ...lastBookmarks])
+        })
+      ])
+    })
+
+    it.each([
+      [undefined],
+      ['neo4j'],
+      ['adb']
+    ])('should not call run updateBookmarks when bookmarks since it didnt have bookmarks', async (metaDb) => {
+      const manager = bookmarkManager({
+        initialBookmarks: new Map([
+          ['neo4j', neo4jBookmarks],
+          ['system', systemBookmarks]
+        ])
+      })
+
+      const updateBookmarksSpy = jest.spyOn(manager, 'updateBookmarks')
+
+      const connection = mockBeginWithSuccess(newFakeConnection())
+
+      const { session } = setupSession({
+        connection,
+        bookmarkManager: manager,
+        beginTx: false,
+        database: 'neo4j'
+      })
+
+      await session.run('query')
```

Comment:
Were in `describe('.beginTransaction()'`. What is this test doing here?

Reply:
I will check it out.